app/util/event_log.py

```diff
@@ -0,0 +1,265 @@
+import collections
+import json
+from logbook import RotatingFileHandler, StreamHandler
+from logbook.more import TaggingHandler, TaggingLogger
+import os
+import sys
+import time
+
+from app.util import fs, log
+from app.util.conf.configuration import Configuration
+from app.util.counter import Counter
+
+
+LOG_CACHE_EXPIRE_TIME_IN_HOURS = 5
+LOG_CACHE_THRESHOLD = 10000
```

Comment:
Sure, why not. I think that would just be a cache on the order of 10s MB at most. We can always drop it later if we see problems.